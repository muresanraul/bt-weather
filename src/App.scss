html, body, div#root {
  width: 100%;
  height: 100%;
  position: relative;
}
html {
  display: table;
  font-size: 10px;
  font-family: 'Roboto', sans-serif;
}
body{
  display: table;
  font-size: 1.8rem;
  background-color: #6dc4ff;
}
div#root{
  display: table-cell;
  vertical-align: middle;
  padding: 1rem;
}
.App {
  display: flex;
  position: relative;
  flex-direction: column;
  max-width: 90rem;
  width: 100%;
  border: #bae3ff 0.2rem solid;
  border-radius: .5rem;
  margin: 0 auto;
  vertical-align: middle;
}
