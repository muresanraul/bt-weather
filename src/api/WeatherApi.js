import {endpoints, formatEndpoint } from '../configurations/';
import RestHelper from './RestHelper';
import { sleep } from '../helpers';

class WeatherApi {

  static async getWeatherData(location) {
    try {
      const response = await RestHelper.get(formatEndpoint(endpoints.weatherApiUrl, {location}));
      debugger;
      return await sleep(response);
    } catch (error) {
      console.error(error);
    }
  }

}

export default WeatherApi;
