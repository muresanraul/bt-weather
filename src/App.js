import React from 'react';

import { WeatherApi } from './api/';
import { WeatherComponent } from './features/WeatherComponent';
import { ProgressComponent } from './features/sharedComponents';
import { AppService } from './services';
import { LOCATION } from './configurations';
import './App.scss';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.refrashDataInterval = null;
    this.state = {
      weatherData: null,
      timeElapsed: 0
    };
  }
  async componentDidMount(){
    this.getWeather();
    this.refrashDataInterval = setInterval(this.refrash, 1000);
  }

  refrash = () => {
    const time = this.state.timeElapsed + 1;
    if (time <= 60) {
      this.setState({timeElapsed: time})
    } else {
      this.setState({timeElapsed: 0}, this.getWeather )
    }
  }

  getWeather = async () => {
    const { data } = await WeatherApi.getWeatherData(LOCATION);
    const weatherData = AppService.getFormatedWeatherData(data)
    this.setState({weatherData})
  }

  componentWillUnmount() {
    clearInterval(this.refrashDataInterval);
  }

  render() {
    return (
      <div className="App">
        <ProgressComponent time={this.state.timeElapsed}/>
        <WeatherComponent data={this.state.weatherData}/>
      </div>
    );
  }
}

export default App;
