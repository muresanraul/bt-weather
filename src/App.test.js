import React from 'react';
import renderer from 'react-test-renderer';

import WeatherApi from './api/WeatherApi';
import App from './App';

jest.mock('./api/WeatherApi');

describe('App:', () => {

  it('Component - renders correctly', () => {
    const dataJson = '{"coord":{"lon":-0.13,"lat":51.51},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03n"}],"base":"stations","main":{"temp":15.36,"feels_like":8.2,"temp_min":14.44,"temp_max":16.67,"pressure":1013,"humidity":51},"visibility":10000,"wind":{"speed":8.7,"deg":260},"clouds":{"all":47},"dt":1593984507,"sys":{"type":1,"id":1414,"country":"GB","sunrise":1593921051,"sunset":1593980346},"timezone":3600,"id":2643743,"name":"London","cod":200}';
    const data = JSON.parse(dataJson);
    WeatherApi.getWeatherData.mockImplementationOnce(() => Promise.resolve(data));
    const app = renderer.create(
      <App/>
    ).toJSON();
    expect(app).toMatchSnapshot();
  });

});