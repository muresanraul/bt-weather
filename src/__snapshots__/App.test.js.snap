// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Component - renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="progress"
  >
    <div
      className="progress__bar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    className="weather"
  >
    <div
      className="LoadingScreen"
    >
      <div
        className="LoadingScreen--background"
      >
        <div
          className="LoadingScreen--message"
        >
          <div
            className="lds-css ng-scope"
          >
            <div
              className="lds-spin"
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <div>
                <div />
              </div>
              <div>
                <div />
              </div>
              <div>
                <div />
              </div>
              <div>
                <div />
              </div>
              <div>
                <div />
              </div>
              <div>
                <div />
              </div>
              <div>
                <div />
              </div>
              <div>
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="weather__title"
    >
      <div>
        <div
          className="dynamic-text"
        >
          <div
            className="dynamic-text__placeholder"
            style={
              Object {
                "backgroundColor": "#ecf8ff",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="weather__main-details"
    >
      <div
        className="weather__main-details-name"
      >
        <div
          className="dynamic-text"
        >
          <div
            className="dynamic-text__placeholder"
            style={
              Object {
                "backgroundColor": undefined,
              }
            }
          />
        </div>
      </div>
      <div
        className="weather__main-details-img"
      >
        <div
          className="image-loader"
        >
          <div
            className="image-loader__placeholder"
          />
        </div>
      </div>
      <div
        className="weather__main-details-explenation"
      >
        <div
          className="dynamic-text"
        >
          <div
            className="dynamic-text__placeholder"
            style={
              Object {
                "backgroundColor": undefined,
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="weather__container"
    >
      <div
        className="weather__column"
      >
        <div
          className="weather__row"
        >
          <div
            className="weather__label"
          >
            Temperature
          </div>
          <div
            className="weather__data"
          >
            <div
              className="dynamic-text"
            >
              <div
                className="dynamic-text__placeholder"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="weather__row"
        >
          <div
            className="weather__label"
          >
            Feels like
          </div>
          <div
            className="weather__data"
          >
            <div
              className="dynamic-text"
            >
              <div
                className="dynamic-text__placeholder"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="weather__row"
        >
          <div
            className="weather__label"
          >
            Minimum temperature
          </div>
          <div
            className="weather__data"
          >
            <div
              className="dynamic-text"
            >
              <div
                className="dynamic-text__placeholder"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="weather__row"
        >
          <div
            className="weather__label"
          >
            Maximum temperature
          </div>
          <div
            className="weather__data"
          >
            <div
              className="dynamic-text"
            >
              <div
                className="dynamic-text__placeholder"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="weather__column"
      >
        <div
          className="weather__row"
        >
          <div
            className="weather__label"
          >
            Presure
          </div>
          <div
            className="weather__data"
          >
            <div
              className="dynamic-text"
            >
              <div
                className="dynamic-text__placeholder"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="weather__row"
        >
          <div
            className="weather__label"
          >
            Humidity
          </div>
          <div
            className="weather__data"
          >
            <div
              className="dynamic-text"
            >
              <div
                className="dynamic-text__placeholder"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="weather__row"
        >
          <div
            className="weather__label"
          >
            Visibility
          </div>
          <div
            className="weather__data"
          >
            <div
              className="dynamic-text"
            >
              <div
                className="dynamic-text__placeholder"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="weather__row"
        >
          <div
            className="weather__label"
          >
            Windspeed
          </div>
          <div
            className="weather__data"
          >
            <div
              className="dynamic-text"
            >
              <div
                className="dynamic-text__placeholder"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
